sphinx/builders/linkcheck.py

```diff
@@ -469,25 +472,31 @@ class HyperlinkCollector(SphinxPostTransform):
     def run(self, **kwargs: Any) -> None:
         builder = cast(CheckExternalLinksBuilder, self.app.builder)
         hyperlinks = builder.hyperlinks
+        to_ignore = [re.compile(x) for x in self.app.config.linkcheck_ignore]
+
+        def add_hyperlink(uri: str, lineno: int) -> None:
+            docname = self.env.docname
+            if any(ignore_re.match(uri) for ignore_re in to_ignore):
+                builder.process_result(CheckResult(uri, docname, lineno, 'ignored', '', 0))
```

Comment:
I feel `HyperlinkCollector` and the builder are strongly coupled. It would be better to separate collecting and processing. How about keeping this transform as a collector of hyperlinks?

We can do the same thing on `CheckExternalLinksBuilder.final()`.
```
def final(self):
    ...

    for hyperlink in self.hyperlinks.values():
        if any(ignore_re.match(hyperlink.uri) for ignore_re in to_ignore):
            self.process_result(hyperlink)
        else:
            self.wqueue.put(hyperlink, False)
```